src/Microsoft.AspNetCore.Routing/Builder/EndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,210 @@
+        {
+            return MapVerbs(builder, pattern, displayName, requestDelegate, GetVerb, metadata);
+        }
+
+        public static IEndpointConventionBuilder MapPost(
+            this IEndpointRouteBuilder builder,
+            string pattern,
+            RequestDelegate requestDelegate,
+            params object[] metadata)
+        {
+            return MapVerbs(builder, pattern, displayName: null, requestDelegate, PostVerb, metadata);
+        }
+
+        public static IEndpointConventionBuilder MapPost(
+            this IEndpointRouteBuilder builder,
+            string pattern,
+            string displayName,
+            RequestDelegate requestDelegate,
+            params object[] metadata)
+        {
+            return MapVerbs(builder, pattern, displayName, requestDelegate, PostVerb, metadata);
+        }
+
+        public static IEndpointConventionBuilder MapPut(
+            this IEndpointRouteBuilder builder,
+            string pattern,
+            RequestDelegate requestDelegate,
+            params object[] metadata)
+        {
+            return MapVerbs(builder, pattern, displayName: null, requestDelegate, PutVerb, metadata);
+        }
+
+        public static IEndpointConventionBuilder MapPut(
+            this IEndpointRouteBuilder builder,
+            string pattern,
+            string displayName,
+            RequestDelegate requestDelegate,
+            params object[] metadata)
+        {
+            return MapVerbs(builder, pattern, displayName, requestDelegate, PutVerb, metadata);
+        }
+
+        public static IEndpointConventionBuilder MapDelete(
+            this IEndpointRouteBuilder builder,
+            string pattern,
+            RequestDelegate requestDelegate,
+            params object[] metadata)
+        {
+            return MapVerbs(builder, pattern, displayName: null, requestDelegate, DeleteVerb, metadata);
+        }
+
+        public static IEndpointConventionBuilder MapDelete(
+            this IEndpointRouteBuilder builder,
+            string pattern,
+            string displayName,
+            RequestDelegate requestDelegate,
+            params object[] metadata)
+        {
+            return MapVerbs(builder, pattern, displayName, requestDelegate, DeleteVerb, metadata);
+        }
+
+        public static IEndpointConventionBuilder MapVerbs(
+           this IEndpointRouteBuilder builder,
+           string pattern,
+           RequestDelegate requestDelegate,
+           IList<string> httpMethods,
+           params object[] metadata)
+        {
+            return MapVerbs(builder, pattern, displayName: null, requestDelegate, httpMethods, metadata);
+        }
+
+        public static IEndpointConventionBuilder MapVerbs(
+           this IEndpointRouteBuilder builder,
+           string pattern,
+           string displayName,
+           RequestDelegate requestDelegate,
+           IList<string> httpMethods,
+           params object[] metadata)
```

Comment:
Why do we have metadata passed in here? Why not instead just make it so that you can add it via `IEndpointConventionBuilder`?

Reply:
Sure metadata could be in an extension method. Do we want it to be an extension method?